docs/user_docs/concepts/package-and-module.md

```diff
@@ -0,0 +1,154 @@
+# KCL Module and Package
+
+This section mainly describes how to organize files in KCL.
+
+## Overview
+
+Within a **module**, KCL organizes files grouped by **package**. A package can be defined within the module or externally (through KCL package manager `kpm`). In the latter case, KCL maintains a copy of the package within the module in a dedicated location.
+
+## Module
+
+A KCL module contains a configuration laid out in a directory hierarchy. It contains everything that is needed to deterministically determine the outcome of a KCL configuration. The root of this directory is marked by containing a `kcl.mod` directory. The contents of this directory are mostly managed by the cue tool. In that sense, `kcl.mod` is analogous to the `.git` directory marking the root directory of a repo, but where its contents are mostly managed by the git tool. Besides, a KCL module is the largest unit of the file organization, has a fixed location of all KCL files and dependencies.
```

Comment:
"The contents of this directory are mostly managed by the cue tool." What is this `cue tool` ？

Reply:
> 

fixed.